e2e/cypress/integration/messaging/receive_message_on_socket_reconnect_spec.js

```diff
@@ -0,0 +1,125 @@
+                        this.wrappedSocket.onopen = this.onopen;
+                        this.wrappedSocket.onmessage = this.onmessage;
+                        this.wrappedSocket.onerror = this.onerror;
+                        this.wrappedSocket.onclose = this.onclose;
+                    },
+                };
+                websockets.push(mockWebSocket);
+                return mockWebSocket;
+            });
+        });
+    });
+
+    it('M18682-RHS fetches messages on socket reconnect when a different channel is in center', () => {
+        // # Connect all sockets
+        websockets.forEach((value) => {
+            value.connect();
+        });
+
+        // # Post a message as another user
+        cy.postMessageAs({sender: users['user-1'], message: 'abc', channelId: townsquareChannelId}).wait(TIMEOUTS.SMALL);
+
+        // # Click "Reply"
+        cy.getLastPostId().then((rootPostId) => {
+            cy.clickPostCommentIcon(rootPostId);
+
+            // # Post a message
+            cy.postMessageReplyInRHS('def');
+
+            // # Save our post for later use
+            cy.getLastPostIdRHS().as('myPostId');
+
+            // # Change channel
+            cy.get('#sidebarItem_suscipit-4').click({force: true}).then(() => {
+                // # Close all sockets
+                websockets.forEach((value) => {
+                    if (value.close) {
+                        value.close();
+                    }
+                });
+
+                // # Post message as a different user
+                cy.postMessageAs({sender: users['user-1'], message: 'ghi', channelId: townsquareChannelId, rootId: rootPostId});
+
+                // # Wait a short time to check whether the message appears or not
+                cy.wait(TIMEOUTS.SMALL);
+
+                // # Get last post Id
+                cy.getLastPostIdRHS().then((beforeWebSocketLastPostId) => {
+                    // * Should be the same post as before
+                    cy.get('@myPostId').should('equal', beforeWebSocketLastPostId);
+
+                    // * Should exist in the RHS
+                    cy.get(`#rhsPost_${beforeWebSocketLastPostId}`).should('exist');
+
+                    // * Connect all sockets one more time
+                    websockets.forEach((value) => {
+                        value.connect();
+                    });
+
+                    // # Wait for sockets to be connected
+                    cy.wait(TIMEOUTS.MEDIUM);
+
+                    // # Get last post Id
+                    cy.getLastPostIdRHS().then((afterWebSocketLastPostId) => {
+                        // * Should be different Id as before
+                        cy.get('@myPostId').should('not.equal', afterWebSocketLastPostId);
+
+                        // * Should exist in RHS
+                        cy.get(`#rhsPost_${afterWebSocketLastPostId}`).should('exist');
+                    });
+                });
```

Comment:
This is good but would be nice if this can be written similar to how the real user sees it.  I'm thinking something like:
```javascript
// * Verify that only "def" is posted and not "ghi"
cy.get('#rhsPostList').should('be.visible').children().should('have.length', 1);
cy.get('#rhsPostList').within(() => {
    cy.findByText('def').should('be.visible');
    cy.queryByText('ghi').should('not.exist');

    // * Connect all sockets one more time
    websockets.forEach((value) => {
        value.connect();
    });
});

// # Wait for sockets to be connected
cy.wait(TIMEOUTS.SMALL);

// * Verify that both "def" and "ghi" are posted on websocket reconnect
cy.get('#rhsPostList').should('be.visible').children().should('have.length', 2);
cy.get('#rhsPostList').within(() => {
    cy.findByText('def').should('be.visible');
    cy.findByText('ghi').should('be.visible');
});
```